ToDoWebApp/Controllers/CategoryController.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using ToDoWebApp.Data.Intefaces;
+using ToDoWebApp.Models;
+
+namespace ToDoWebApp.Controllers
+{
+    public class CategoryController : Controller
+    {
+
+        public static List<Category> asList;
+        public static bool pereita;
+
+        public CategoryController(ICategoryRepository IcategoryRepository)
+        {
+            if (pereita.Equals(false))
+            {
+                asList = IcategoryRepository.categories.ToList();
+                pereita = true;
+
+            }
+        }
+
+
+
+        // GET: CategoryController
+        public ActionResult Index()
+        {
+            return View(asList);
+        }
+
+        // GET: CategoryController/Details/5
+        public ActionResult Details(int id)
+        {
+            return View(model: asList.FirstOrDefault(predicate: s => s.CategoryId == id));
+        }
+
+        // GET: CategoryController/Create
+        public ActionResult Create()
+        {
+            return View(new Category { Name = "Type Here" });
+        }
+
+        // POST: CategoryController/Create
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public ActionResult Create(Category category)
+        {
+            try
+            {
+                // duodame prieiga prie duom
+                asList.Add(category);
+                return RedirectToAction(nameof(Index));
+            }
+            catch
+            {
+                return View(category);
+            }
+        }
+
+        // GET: CategoryController/Edit/5
+        public ActionResult Edit(int id)
+        {
+            return View();
```

Comment:
I guess after you push Edit you get empty form. Try passing something here to fill form with item data you want to alter